packages/shared/components/inputs/Input.svelte

```diff
@@ -78,39 +83,39 @@
         }
     }
 
-    const onPaste = (e) => {
-        if (e.clipboardData && (float || integer)) {
-            const pasteVal = e.clipboardData.getData('text')
+    function onPaste(event: ClipboardEvent): void {
+        if (event.clipboardData && (float || integer)) {
+            const pasteVal = event.clipboardData.getData('text')
             // Discard scientific notation or negative
             if (pasteVal.indexOf('e') >= 0 || pasteVal.indexOf('-') >= 0) {
-                e.preventDefault()
+                event.preventDefault()
             } else if (float) {
                 const val = parseCurrency(pasteVal)
                 // Discard any numbers we can't parse as floats
                 if (Number.isNaN(val)) {
-                    e.preventDefault()
+                    event.preventDefault()
                 } else if (maxDecimals !== undefined) {
                     value = formatNumber(val, undefined, maxDecimals, 0)
-                    e.preventDefault()
+                    event.preventDefault()
                 }
             } else if (integer) {
                 // Dicard anything with a decimal separator
                 if (allDecimalSeparators.some((sep) => pasteVal.indexOf(sep) >= 0)) {
-                    e.preventDefault()
+                    event.preventDefault()
                 } else {
                     const val = Number.parseInt(pasteVal, 10)
                     // Discard any number we can't parse as integers
                     if (Number.isNaN(val)) {
-                        e.preventDefault()
+                        event.preventDefault()
                     }
                 }
             }
         }
     }
```

Comment:
Could use some refactoring, very complicated flow. `event.preventDefault()` is called a lot of times. I think we're using a block list here instead of an allowlist. Maybe switching this would make the logic easier to comprehend?